src/mentat/clerk_utils/build.clj

```diff
@@ -0,0 +1,191 @@
+(ns mentat.clerk-utils.build
+  (:require [babashka.fs :as fs]
+            [clojure.java.shell :refer [sh]]
+            [clojure.string]
+            [hiccup.page :as hiccup]
+            [mentat.clerk-utils.shadow :as shadow]
+            [nextjournal.clerk :as clerk]
+            [nextjournal.clerk.config :as config]
+            [nextjournal.clerk.view]
+            [nextjournal.clerk.viewer :as cv]))
+
+(def ^:no-doc custom-css
+  (atom []))
+
+(defn- rebind [^clojure.lang.Var v f]
+  (let [old (.getRawRoot v)]
+    (.bindRoot v (f old))))
+
+(defonce _ignore
+  (rebind
+   #'nextjournal.clerk.view/include-viewer-css
+   (fn [old]
+     (fn [& xs]
+       (concat
+        (map hiccup/include-css @custom-css)
+        (apply old xs))))))
+
+(defn reset-css! []
+  (reset! custom-css []))
+
+(defn add-css! [& entries]
+  (swap! custom-css into entries))
+
+;; TODO
+
+;; - test the watcher, see if local file changes get picked up
+;; -
+
+(defn git-sha
+  "Returns the sha hash of this project's current git revision."
+  []
+  (-> (sh "git" "rev-parse" "HEAD")
+      (:out)
+      (clojure.string/trim)))
+
+(defn git-dependency
+  "Nice for display!"
+  [slug]
+  (clerk/md
+   (format
+    "```clj
+
+{'io.github.%s
+  {:git/sha \"%s\"}}
+  ```" slug (git-sha))))
+
+(defn replace-sha-template!
+  "Given some `path`, modifies the file at `path` replaces any occurence of the
+  string `$GIT_SHA` with the actual current sha of the repo."
+  ([path]
+   (replace-sha-template! path (git-sha)))
+  ([path sha]
+   (-> (slurp path)
+       (clojure.string/replace "$GIT_SHA" sha)
+       (->> (spit path)))))
+
+(def ^:private js-k "/js/viewer.js")
+
+(defn viewer-js []
+  (@config/!resource->url js-k))
+
+(defn set-viewer-js! [path]
+  (swap! config/!resource->url assoc js-k path))
+
+(defn reset-viewer-js! []
+  (let [default (@config/!asset-map js-k)]
+    (swap! config/!resource->url assoc js-k default)))
+
+(defn with-viewer-js [path f]
+  (let [v (viewer-js)]
+    (try (set-viewer-js! path)
+         (f)
+         (finally
+           (set-viewer-js! v)))))
+
+(defn slug->url [slug]
+  (str "https://github.com/" slug))
+
+(defn replace-sha [s sha]
+  (if (string? s)
+    (clojure.string/replace s "$GIT_SHA"  sha)
+    s))
+
+(defn get-subfolder [{:keys [github-slug garden? subfolder]} sha]
+  (-> (or subfolder
+          (when (and garden? github-slug)
+            (str github-slug "/commit/$GIT_SHA/")))
+      (replace-sha sha)))
+
+;; API Clones
+
+(defn serve!
+  "Like `clerk/serve!` but handles custom cljs. If `browse?` and `show` are true,
+  shows the entry at `show!` once it opens."
+  [{:keys [cljs-namespaces browse? index] :as opts}]
+  (when (seq cljs-namespaces)
+    (let [{:keys [js-path]} (shadow/watch! cljs-namespaces)]
+      (set-viewer-js! js-path)))
+  (try (clerk/serve! opts)
+       (finally
+         (when (and browse? index)
+           (Thread/sleep 500)
```

Comment:
Curious why the sleep is needed, something we should fix upstream maybe?

Reply:
looks like it's not, I've removed it. Good catch!